File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.py
class Solution:
    def isFascinating(self, n: int) -> bool:
        m = n
        cnt= 0
        arr = []
        while(m):
            if(m%10  in arr):
                return False
            else:
                arr.append(m%10)
            m//=10;
        m = 2*n
        while(m):
            if(m%10  in arr):
                return False
            else:
                arr.append(m%10)
            m//=10;
        m = 3*n
        while(m):
            if(m%10  in arr):
                return False
            else:
                arr.append(m%10)
            m//=10;
        # print(len(arr))
        if(0 in arr):
            return False
        return len(arr)==9

File: 2729-check-if-the-number-is-fascinating/2729-check-if-the-number-is-fascinating.cpp
class Solution {
public:
    bool isFascinating(int n) {
        int n1=n+n,n2=3*n;
        cout<<n1<<" "<<n2;
        set<int>s;
        int cnt=0;
        while(n){
            s.insert(n%10);
            n/=10;
            ++cnt;
        }
        n=n1;
        while(n){
            s.insert(n%10);
            n/=10;
            ++cnt;
        }
        n=n2;
        while(n){
            s.insert(n%10);
            n/=10;
            ++cnt;
        }
        s.insert(0);
        if(s.size()==10 and cnt==9) return true;
        return false;
    }
};